test/system/projects/samples_test.rb

```diff
@@ -41,1291 +199,1535 @@ class SamplesTest < ApplicationSystemTestCase
+      within('#samples-table table tbody') do
+        assert_text 'New Name'
+      end
+      ### VERIFY END ###
     end
 
-    test 'should update Sample' do
+    test 'edit sample' do
+      ### SETUP START ###
       visit namespace_project_sample_url(@namespace, @project, @sample1)
-      assert_selector 'a', text: I18n.t('projects.samples.show.edit_button'), count: 1
-      click_on I18n.t('projects.samples.show.edit_button'), match: :first
+      ### SETUP END ###
 
-      fill_in 'Description', with: @sample1.description
+      ### ACTIONS START ###
+      # nav to edit sample page
+      click_on I18n.t('projects.samples.show.edit_button')
+
+      # verify current sample doesn't have with new properties that will be used
+      assert_no_text 'A new description'
+      assert_no_text 'New Sample Name'
+      # change current sample properties
+      fill_in 'Description', with: 'A new description'
       fill_in 'Name', with: 'New Sample Name'
       click_on I18n.t('projects.samples.edit.submit_button')
+      ### ACTIONS END ###
 
+      ### results start ###
+      # success flash msg
       assert_text I18n.t('projects.samples.update.success')
-      assert_text 'New Sample Name'
-      assert_text @sample1.description
+
+      # verify redirect to sample show page and updated sample state
+      assert_selector 'h1', text: 'New Sample Name'
+      assert_text 'A new description'
+      ### results end ###
     end
 
-    test 'should destroy Sample from sample show page' do
+    test 'destroy sample from sample show page' do
+      ### SETUP START ###
+      # nav to samples index and verify sample exists within table
+      visit namespace_project_samples_url(@namespace, @project)
+      assert_selector "#samples-table table tbody tr[id='#{@sample1.id}']"
+      assert_selector '#samples-table table tbody tr', count: 3
+
+      # nav to sample show
       visit namespace_project_sample_url(@namespace, @project, @sample1)
-      assert_link text: I18n.t('projects.samples.index.remove_button'), count: 1
+      ### SETUP END ###
+
+      ### ACTIONS START ##
+      # remove sample
       click_link I18n.t(:'projects.samples.index.remove_button')
 
       within('#turbo-confirm[open]') do
         click_button I18n.t(:'components.confirmation.confirm')
       end
+      ### ACTIONS END ###
 
+      ### VERIFY START ###
+      # success flash msg
       assert_text I18n.t('projects.samples.deletions.destroy.success', sample_name: @sample1.name,
                                                                        project_name: @project.namespace.human_name)
-
-      assert_no_selector '#samples-table table tbody tr', text: @sample1.name
+      # redirected to samples index
       assert_selector 'h1', text: I18n.t(:'projects.samples.index.title'), count: 1
+      # deleted sample row no longer exists
+      assert_no_selector "#samples-table table tbody tr[id='#{@sample1.id}']"
```

Comment:
**nitpick (blocking):** Can you move this below the other assertion as that would ensure that the samples table has been loaded.

Reply:
Fixed in d60b9028eeca99d5aad54b37f1e7ce872ceb8dac